### ISSUE TITLE
Ansible warnings: potential error in future Ansible release

### ISSUE THREAD

**Issue by hackermd** [gc3-uzh-ch/elasticluster#443]
With Ansible version `2.3.0`, there are multiple warnings related to syntax of boolean expressions for `when` statements:

```
[WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}
```

The warning may indicate that this will raise an error in future releases.

**Comment by belambert** [comment]
Is this a real problem with a potential fix?  I get TONS of this warning.  It drowns out the other messages and is really annoying!

**Comment by riccardomurri** [comment]
They are warnings about ElastiCluster (ab)using Jinja template syntax in Ansible's `where:` clauses.  This will become an actual issue when (if) Ansible makes some change and stops supporting Jinja template expansion in `where:` clauses, but as Ansible 2.3 this is only just a "heads up" about potential future incompatibility.

Yes, they're annoying but it's somewhat delicate to fix (as I basically have to check and potentially rewrite almost every conditional expression in the playbooks) so I expect to break things when I start to work on it, and I'd rather have release 1.3 out before attempting to fix it.

If anyone is willing to try submit a PR, I will be glad to provide guidance and hints on what to change.

**Comment by belambert** [comment]
Excellent, thank you for the explanation.

**Comment by riccardomurri** [comment]
Fixed in commits ab3fa1c through d3cc5f9.
